### PR TITLE
add more documentation about how tests work

### PR DESCRIPTION
**Pull request by dothebart**

### CODE REVIEW

File: README_maintainers.md [hunk 1]
```diff
@@ -16,7 +16,12 @@ Main sections:
   - [Linux Core Dumps](#linux-core-dumps)
   - [Windows Core Dumps](#windows-core-dumps)
 - [Unittests](#unittests)
-
+  - [invoking driver tests](#driver-tests) via scripts/unittests
+  - [capturing test communication](#running-tcpdump--windump-for-the-sut) but
```
Comment (jsteemann) [inline README_maintainers.md:20]: I would rephrase this as "capturing test HTTP communication".
Additionally, is the "--" in the anchor intentional?

File: README_maintainers.md [hunk 2]
```diff
@@ -16,7 +16,12 @@ Main sections:
...
+    [maybe also downgrade to JSON](#forcing-downgrade-from-vpack-to-json)
```
Comment (jsteemann) [inline README_maintainers.md:21]:
```suggestion
    [forcing communication to use plain-text JSON](#forcing-downgrade-from-vpack-to-json)
```

File: tests/README.md [hunk 3]
```diff
@@ -0,0 +1,194 @@
+# Tests
+When creating tests, please bear in mind, that every test added will add additional time during the PR-Runs.
+
+So every test added *will* slow down PRs.
```
Comment (jsteemann) [inline tests/README.md:4]:
```suggestion
When creating tests, please bear in mind, that every test added will add additional time during the PR runs.

So every added test *will* slow down PR runs.
```

File: tests/README.md [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+
+Please also note that tests in the CI share a system with other runs in parallel. 
+So CI tests are not a place to max out I/O, Memory, or such.
```
Comment (jsteemann) [inline tests/README.md:7]:
```suggestion
So CI tests are not a place to max out I/O, memory, or such.
```

File: tests/README.md [hunk 5]
```diff
@@ -0,0 +1,194 @@
+# Tests
+When creating tests, please bear in mind, that every test added will add additional time during the PR-Runs.
+
+So every test added *will* slow down PRs.
+
+Please also note that tests in the CI share a system with other runs in parallel.
```
Comment (jsteemann) [inline tests/README.md:6]:
```suggestion
Please also note that test runs in the CI share the system with other runs in parallel. 
```

File: tests/README.md [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+When creating tests, please bear in mind, that every test added will add additional time during the PR-Runs.
+
+So every test added *will* slow down PRs.
+
+Please also note that tests in the CI share a system with other runs in parallel. 
+So CI tests are not a place to max out I/O, Memory, or such.
+
+So please use *BULK* inserts to slimline the creation of your test data.
+If you can use a common dataset for several tests, please use `setUpAll` and `tearDownAll` to reduce resource usage.
+If splitting a testsuite between read only and writing tests is neccessary for this, please do so.
```
Comment (jsteemann) [inline tests/README.md:11]:
```suggestion
If splitting a testsuite between read only and writing tests is necessary for this, please do so.
```

File: tests/README.md [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+
+### Testsuites
+Testsuites live in `js/client/modules/@arangodb/testsuites/`. This folder is spidered during the startup of `testing.js` and loads the files as plugins.
```
Comment (jsteemann) [inline tests/README.md:14]:
```suggestion
Testsuites live in `js/client/modules/@arangodb/testsuites/`. This folder is spidered during the startup of `testing.js` and all files in it are loaded automatically.
```

File: tests/README.md [hunk 8]
```diff
@@ -0,0 +1,194 @@
...
+Please think twice, whether you need to create a new testsuite, or your test will fit into one of the existing ones. Suites need to be registered within Oskar.
```
Comment (jsteemann) [inline tests/README.md:15]:
```suggestion
Please think twice whether you need to create a new testsuite or your test will fit into one of the existing ones. All test suites need to be registered within Oskar in order to be executed during CI tests.
```

File: tests/README.md [hunk 9]
```diff
@@ -0,0 +1,194 @@
...
+The Plugin structure works as follows:
```
Comment (jsteemann) [inline tests/README.md:16]:
```suggestion
The structure of each testsuite is as follows: 
```

File: tests/README.md [hunk 10]
```diff
@@ -0,0 +1,194 @@
...
+  // if your testsuite has CLI parameters, specify them including the default param here:
+  opts['skipAql'] = false;
+
+  // 
+  for (var attrname in functionsDocumentation) { fnDocs[attrname] = functionsDocumentation[attrname]; }
+  for (var i = 0; i < optionsDocumentation.length; i++) { optionsDoc.push(optionsDocumentation[i]); }
+};
+```
+
+### existing Javascript tests
```
Comment (jsteemann) [inline tests/README.md:73]:
```suggestion
### Existing Javascript tests
```

File: tests/README.md [hunk 11]
```diff
@@ -0,0 +1,194 @@
...
+
+#### [permission] => tests/js/client/permissions / [server_parameters] => tests/js/client/server_parameters / [server_permissions] => client/server_permissions
+This set of testsuites is here to revalidate startup parameters of arangodb processes.
```
Comment (jsteemann) [inline tests/README.md:76]:
```suggestion
This set of testsuites is here to revalidate startup parameters of arangod processes.
```

File: tests/README.md [hunk 12]
```diff
@@ -0,0 +1,194 @@
...
+The dump family of testsuites lives in `js/client/modules/@arangodb/testsuites/dump.js`
+and orchestrates a sequence:
+- launching a SUT
+- create test data within it
+- dump this test data
+- execute some manual sanity checks on the dump files
+- restore it to a fresh SUT
+- check data
+
+Since many create & check functions should be common to *all* suites, the default behavior is to enable them, and only blacklist them out if unsupported.
```
Comment (jsteemann) [inline tests/README.md:109]:
```suggestion
Since many create & check functions should be common to *all* suites, the default behavior is to enable them, and only exclude them if unsupported.
```

File: tests/README.md [hunk 13]
```diff
@@ -0,0 +1,194 @@
...
+So `tests/js/server/dump/dump-setup[-cluster].js` will be invoked during setting up the first SUT.
+It derives all its functions from `dump-setup-common.inc` and only invokes them in sequence.
+
+Revalidating will be done using `dump[-cluster].js`, which will obtain all its test functions from `dump-test.inc`.
+
+Testdata to be used during the first create etc can be put into `tests/js/server/dump`.
+
+
+#### [communication, chaos] testsuite to launch parallel load
+These testsuites are intended to launch several arangosh, in order to generate
```
Comment (jsteemann) [inline tests/README.md:133]:
```suggestion
These testsuites are intended to launch several arangosh instances, in order to generate concurrent requests.
```

File: README_maintainers.md (hunk 1 above)
Comment (dothebart) [reply README_maintainers.md:20]: `### Running tcpdump / windump for the SUT` is probably converted to `--` by the github headline anchor generator.
